### ISSUE TITLE
Issue with TotalItems

### ISSUE THREAD

**Issue by panzea** [michaelbromley/ngx-pagination#151]
Hi, 

I am having an issue with the totalItems property of the pagination, when retrieving data from an agular2 service that pulls information from a REST api.

My component html is as follows:
```
 <tr *ngFor=' let store of stores | paginate : { itemsPerPage: storesPerPage, currentPage: currentPage, id : 1, totalItems: totalItems } ' >
                <td>{{ store.storeId }}</td>
                <td>{{ store.name }}</td>
                <td>{{ store.address.street1 }} {{ store.address.zip }}</td>
                <td>{{ store.phone }}</td>
              </tr>
              </tbody>
            </table>

            <pagination-controls
              (pageChange) = "currentPage = populateStores($event)"
              id="1"
              >
            </pagination-controls>
```
The relevant parts of my component JS is:

```
stores = [];
  storesPerPage = 100;
  totalItems = 0;
  currentPage = 1;

  constructor(
    private storesService: StoresService
  ) {
    console.log('Construct');
    this.populateStores(1);
  }


  populateStores(pageNumber) {
    this.storesService.getAll(this.storesPerPage, pageNumber)
      .subscribe(response => {
        if(response.status === 'error') {
          console.log( 'Error response received' );
        } else {
          this.totalItems = response.totalItems;
          this.stores = response.stores;
        }

      },
      error => {
        console.log('Server error');
      });

    return pageNumber;
  }
```

The storesService.gatAll command returns an object that looks like:
```
{
status: "success",
totalItems: 1500,
stores: [
  { ... store object ... },
  { ... store object ... },
  { ... store object ... },
  etc.
]
}
```

I am expecting that when I update the components this.totalItems property, that the pagination pipe in the ngFor of the component would receive the new value, and the pagination would display as expected. However this doesn't seem to happen. Instead, it assumes that totalItems is 0, and displays no pagination (just greyed out previous/next buttons)

If I update the "totalItems = 0" to "totalItems = 400" at the start of the component JS, this binds as expected and the pagination displays correctly (but only shows pagination for 400 results). Equally, if I update the ngFor pagination pipe to read "paginate : { itemsPerPage: storesPerPage, currentPage: currentPage, id : 1, totalItems: 400 } " it will display pagination correctly for the first 400 results.

What am I missing that is preventing me from binding the correct totalItems (as retrieved from my service) to the ngFor pagination pipe?

P.S. I have confirmed that my service is retrieving a valid totalItems value, and have also tried hardcoding this too to ensure that this wasn't the problem, but with the same outcome

I'm using v3.0.0 of the component..

Any help greatly appreciated

Gary

**Comment by michaelbromley** [comment]
Hi Gary,

Nothing jumps out as being obviously wrong with the code you posted there. What do you get is you `console.log(response)` in the subscribe function after calling `getAll()`?

**Comment by panzea** [comment]
If I add a console.log(response.totalItems); in the subscribe function of getAll() it writes out the correct number in the console (in this case 15) representing the 15 objects, but it doesn't seem to get recongised by the paging-controls control, and it just looks the same as if it had been sent to 0 (which is what it's set to on initialisation)

As far as I can see, setting this.totalItems to a positive number should get the paging-controls to update and render the new page links (depending on what the totalItems variable is set to and what the storesPerPage variable is set to). I've even tried hardcoding the this,totalItems = 300 in the subscribe() method just in case it required at least one pages worth of results before it did anything, but this makes no difference either.

Any other ideas?

**Comment by panzea** [comment]
Actually, I think I fixed the problem. There are one of a few things that fixed it, so I'm not sure which did it in the end, but hopefully this will help someone if they find themselves in a similar situation:

1. In my original example the "id" applied to the ng-For was listed as id: 1, whereas the id of the pagination controls was listed as id: "1". I suspect that this discrepancy is what was causing the problem.

2. The other thing I changed was to change my data source for the ngFor into an observable<object[]> and use the async pipe before the paginate pipe. I don't think this really fixed the problem, but seems like a better pattern. I found this after reading through the Server Side Paging demo at http://michaelbromley.github.io/ngx-pagination/#/server-paging.

Thanks for your help on this though.
